### ISSUE TITLE
THCHS-30 ali-to-pdf crash

### ISSUE THREAD

**Issue by kckao** [kaldi-asr/kaldi#2418]
```
steps/nnet/train.sh --copy_feats false --cmvn-opts --norm-means=true --norm-vars=false --hid-layers 4 --hid-dim 1024 --learn-rate 0.008 data/fbank/train data/fbank/dev data/lang exp/tri4b_ali exp/tri4b_ali_cv exp/tri4b_dnn

# INFO
steps/nnet/train.sh : Training Neural Network
         dir       : exp/tri4b_dnn
         Train-set : data/fbank/train 10000, exp/tri4b_ali
         CV-set    : data/fbank/dev 893 exp/tri4b_ali_cv

LOG ([5.4.129~1-90363]:main():cuda-gpu-available.cc:49)

# PREPARING ALIGNMENTS
Using PDF targets from dirs 'exp/tri4b_ali' 'exp/tri4b_ali_cv'
hmm-info exp/tri4b_ali/final.mdl
steps/nnet/train.sh: line 177: 31603 Aborted                 (core dumped) analyze-counts --verbose=1 --binary=false --counts-dim=$num_pdf ${frame_weights:+ "--frame-weights=$frame_weights"} ${utt_weights:+ "--utt-weights=$utt_weights"} "$labels_tr_pdf" $dir/ali_train_pdf.counts 2> $dir/log/analyze_counts_pdf.log
# Accounting: time=2 threads=1
# Ended (code 134) at Sun May 13 21:54:59 CST 2018, elapsed time 2 seconds
run.pl: job failed, log is in exp/tri4b_dnn/log/train_nnet.log
+ exit 1
```

**Comment by danpovey** [comment]
thanks-- to fix it I'd need to know more info about how it crashes.
you can head the file $dir/log/analyze_counts_pdf.log to get the exact
command line, then do
gdb --args analyze-counts [args]
(gdb) r
.. wait for it to crash
(gdb) bt  # to get backtrace
(gdb) up # keep going up till you get to an interesting stack frame; 'down'
reverses it.
(gdb) p x  # x is an example: print any relevant-looking variables

**Comment by kckao** [comment]
I'm sorry. That is because OOM.
I look into the analyze_counts_pdf.log, and when it did "ali-to-pdf exp/tri4b_ali/final.mdl 'ark:gunzip -c exp/tri4b_ali/ali.*.gz |' ark:-",
there is  "ERROR:~SequentialTableReaderArchiveImpl():util/kaldi-table-inl.h:678) TableReader: error detected closing archive"
And I found there are a lot of "error detected closing archive" in other logs.
After modified n jobs from 4 to 2,  it was fixed.
